src/qs8-gemm/MRx16c8-avx512vnni.c.in

```diff
@@ -100,126 +126,164 @@ void xnn_${DATATYPE_SPEC}_gemm_minmax${REQUANTIZATION_SPEC}_ukernel_${MR}x16c8__
         __m512i vsum${M}x0123456789ABCDEF = _mm512_mullo_epi32(vksum0123456789ABCDEF, vinput_zero_point${M});
         __m512i vacc${M}x01234567 = _mm512_cvtepu32_epi64(_mm512_extracti64x4_epi64(vsum${M}x0123456789ABCDEF, 0));
         __m512i vacc${M}x89ABCDEF = _mm512_cvtepu32_epi64(_mm512_extracti64x4_epi64(vsum${M}x0123456789ABCDEF, 1));
+    $elif DATATYPE == "QB4":
+      const __m512 vksum0123456789ABCDEF = _mm512_load_ps(w);
+      $for M in range(MR):
+        __m512 vscaled${M}x0123456789ABCDEF = _mm512_mul_ps(vksum0123456789ABCDEF, vinput_zero_point${M});
     $else:
       __m512i vacc0x01234567 = _mm512_cvtepu32_epi64(_mm256_load_si256((const __m256i*) w));
       __m512i vacc0x89ABCDEF = _mm512_cvtepu32_epi64(_mm256_load_si256((const __m256i*) ((const int32_t*) w + 8)));
       $for M in range(1, MR):
         __m512i vacc${M}x01234567 = vacc0x01234567;
         __m512i vacc${M}x89ABCDEF = vacc0x89ABCDEF;
-    $if MR < 4:
-      $for M in range(MR):
-        __m512i vacc1x${M}x01234567 = _mm512_setzero_epi32();
-        __m512i vacc1x${M}x89ABCDEF = _mm512_setzero_epi32();
+    $if DATATYPE != "QB4":
+      $if MR < 4:
+        $for M in range(MR):
+          __m512i vacc1x${M}x01234567 = _mm512_setzero_epi32();
+          __m512i vacc1x${M}x89ABCDEF = _mm512_setzero_epi32();
     w = (const int32_t*) w + 16;
 
-    size_t k = kc;
-    while (k >= 16 * sizeof(int8_t)) {
+    $if DATATYPE == "QB4":
+      $SET_INDENT(1)
+      for (size_t kb=0; kb < kc; kb+=bl) {
```

Comment:
space around operators:
for (size_t kb = 0; kb < kc; kb += bl) {

Reply:
cc @GregoryComer for #6748